src/sisl/physics/electron.py

```diff
@@ -1571,9 +1579,41 @@ def norm2(self, sum=True):
         # Retrieve the overlap matrix (FULL S is required for NC)
         S = self.Sk()
 
-        if sum:
+        # ensure backwards compatibility by converting bool
+        if projection is True:
+            projection = "sum"
+        elif projection is False:
+            projection = "orbital"
+
+        if projection == "sum":
             return self.inner(matrix=S)
-        return conj(self.state) * S.dot(self.state.T).T
+
+        elif projection == "orbital":
+            return conj(self.state) * S.dot(self.state.T).T
+
+        elif projection == "atom":
+            na = self.parent.na
+            no = self.parent.no
```

Comment:
I guess this would break for `Spin(nc)`. since orbitals does not encompass spin.
A better estimate would be `len(self.parent)` which would work when `parent` is a `Hamiltonian` (still failing if the `parent=Geometry` and `state` is from an `NC` calculation. But...

Reply:
changed